packages/mask/src/plugins/Foundation/SNSAdaptor/FoundationProvenances.tsx

```diff
@@ -0,0 +1,75 @@
+import { Typography, Link, Box, Divider, Grid } from '@mui/material'
+import { useAccount } from '@masknet/web3-shared-evm'
+import { useI18N } from '../../../utils'
+import type { NftHistory } from '../types'
+import { convertDate } from '../utils'
+import OpenInNewIcon from '@mui/icons-material/OpenInNew'
+
+interface Props extends React.PropsWithChildren<{}> {
+    histories: NftHistory[]
+}
+
+const formatAddress = (address: string) => `${address.slice(0, 6)}...${address.slice(-4)}`
+
+function FoundationProvenances(props: Props) {
+    const { t } = useI18N()
+    const account = useAccount()
+    props.histories.sort((first: NftHistory, second: NftHistory) => {
+        if (first.date < second.date) {
+            return 1
+        }
+        if (first.date > second.date) {
+            return -1
+        }
+        return 0
+    })
```

Comment:
Sorting an array is a side-effect. Therefore, please create a duplication before sorting it.